extra_geom/detectors.py

```diff
@@ -854,6 +854,276 @@ def to_distortion_array(self, allow_negative_xy=False):
         return super().to_distortion_array(allow_negative_xy)
 
 
+class AGIPD_500KGeometry(DetectorGeometryBase):
+    """Detector layout for AGIPD-500k
+
+    The coordinates used in this class are 3D (x, y, z), and represent metres.
+
+    You won't normally instantiate this class directly:
+    use one of the constructor class methods to create or load a geometry.
+    """
+    detector_type_name = 'AGIPD-500K2G'
+    pixel_size = 2e-4  # 2e-4 metres == 0.2 mm
+    frag_ss_pixels = 64
+    frag_fs_pixels = 64
+    expected_data_shape = (8, 512, 128)
+    n_modules = 8
+    n_tiles_per_module = 16
+
+    @classmethod
+    def from_center(cls, center, asic_gap=2, panel_gap=(16, 30), unit=pixel_size):
+        """Generate an AGIPD-500K2G geometry from center position.
+
+        This produces an idealised geometry, assuming all modules are perfectly
+        flat, aligned and equally spaced within their quadrant.
+
+        The center position is given in pixel units, referring to the first
+        pixel of the first module, corresponding to data channels 0.
+
+        The origin of the coordinates is the bottom-right corner of the
+        detector. Coordinates increase upwards and to the left (looking along
+        the beam).
+
+        To give positions in units other than pixels, pass the *unit* parameter
+        as the length of the unit in metres. E.g. ``unit=1e-3`` means the
+        coordinates are in millimetres.
+        """
+        asic_gap_px = asic_gap * unit / cls.pixel_size
+        panel_gap_x = panel_gap[0] * cls.pixel_size
+        panel_gap_y = panel_gap[1] * cls.pixel_size
+        # print(f'mod gap: ({panel_gap_x}, {panel_gap_y})')
+        # print(f'asic gap: {asic_gap_px}')
+
+        # How much space one tile takes up, including the gaps
+        # separating it from its neighbour.
+        # In the y dimension, (2 * 64) px + gap between modules
+        module_height = 2 * cls.frag_fs_pixels * cls.pixel_size
+        # In x, 64 px + gap between tiles (asics)
+        tile_width = (cls.frag_ss_pixels + asic_gap_px) * cls.pixel_size
+        module_width = 8 * tile_width
+
+        # print(f'module height: {module_height}')
+        # print(f'tile width: {tile_width}')
+        # print(f'module width: {module_width}')
+
+        modules = []
+        for p in range(8):
+            panel_corner_y = (center[1] * unit) - ((p // 2) * (module_height + panel_gap_y))
+            panel_corner_x = (center[0] * unit) + ((p % 2) * (module_width + panel_gap_x))
+            # print(f'panel{p} ({1000 * panel_corner_x:.2f}, {1000 * panel_corner_y:.2f})')
+
+            tiles = []
+            modules.append(tiles)
+
+            for a in range(16):
+                # upper ASIC row is rotated by 180 degre
```

Comment:
It looks like this is going to be fixed upstream (either in AGIPD or the DAQ), so we won't need this in the future. It's not clear yet whether we'll need it as an option for the existing data.

Reply:
yep, I'll stop working on this for now. I think this won't be necessary if the shape is fixed before the first experiment at HED